libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1333,6 +1333,35 @@ string YulUtilFunctions::allocateMemoryArrayFunction(ArrayType const& _type)
 
 string YulUtilFunctions::conversionFunction(Type const& _from, Type const& _to)
 {
+	if (_from.category() == Type::Category::Function)
```

Comment:
We could still think about extracting this to some function or to have some nice multi-dispatch mechanism for this in general, but I'd say it's fine for now...